### ISSUE TITLE
Github Actions environment variables are not auto-detected by the scanner

### ISSUE THREAD

**Issue by pcolmer** [mc1arke/sonarqube-community-branch-plugin#523]
**Describe the bug**
1. Performing a straightforward analysis on a branch commit does not store the analysis under that branch's name. Instead, it gets saved under the name "develop".
2. Performing a pull request does not store the analysis under the "Pull Request" tab on Sonar. Instead, it gets saved against the name "develop".

**To Reproduce**
1. Create an empty database on PostgreSQL. I'm using AWS RDS for this purpose.
2. Start up a Docker container with the image that embeds the plugin. For the purpose of gaining a public IP address, I'm using the facility within Docker Desktop to spin up the container within ECS Fargate but I have no reason to believe that this is changing the behaviour of the container.
3. Perform a GitHub Actions-triggered scan on a branch commit
4. Perform a GitHub Actions-triggered scan on a pull request

**Expected behavior**
1. I would expect the "+" symbol next to the branch name in the Sonar UI to give me a list of branches and not a pop-up explaining how to analyse branches.
2. I would expect commit-triggered analyses to be saved under the correct branch name.
3. I would expect PR-triggered analyses to be saved under the correct branch name and handled as a pull request.

**Software Versions**
 - SonarQube Version: Version 9.2.3 (build 50713)
 - Plugin Version: 1.10.0 (mc1arke/sonarqube-with-community-branch-plugin:9.2-community)

**Additional context**
docker-compose file used to run the Docker container:
```
version: "3"

# Specify the VPC ID within the AWS account
x-aws-vpc: "vpc-12345678"

# Define the Sonarqube service.
services:
  sonarqube:
    image: mc1arke/sonarqube-with-community-branch-plugin:9.2-community
    deploy:
      replicas: 1
      resources:
        limits:
          cpus: '0.5'
          memory: 4096M
    ulimits:
      nofile:
        soft: 65536
        hard: 65536
    environment:
      - sonar.jdbc.url=jdbc:postgresql://mydatabase.us-east-1.rds.amazonaws.com:5432/sonar
      - sonar.jdbc.username=sonar
      - sonar.jdbc.password=sonarpassword
      - sonar.search.javaAdditionalOpts=-Dnode.store.allow_mmap=false
    volumes:
      - sonarqube_data:/opt/sonarqube/data
      - sonarqube_logs:/opt/sonarqube/logs
      - sonarqube_conf:/opt/sonarqube/conf
    ports:
      - target: 9000
        x-aws-protocol: http

# Define the certificate to be used for SSL
x-aws-cloudformation:
  Resources:
    Sonarqube9000Listener:
      Properties:
        Certificates:
          - CertificateArn: "arn:aws:acm:us-east-1:account:certificate/guid"
        Protocol: HTTPS
        Port: 443

# Force the default VPC SG to be used so that access to the database is possible
networks:
  default:
    external: true
    name: sg-12345678

volumes:
  sonarqube_data:
  sonarqube_logs:
  sonarqube_conf:
```

Steps in the GitHub Actions workflow:
```
      - name: SonarQube scan
        uses: sonarsource/sonarqube-scan-action@master
        env:
          SONAR_TOKEN: ${{ secrets.SONARQUBE_TOKEN }}
          SONAR_HOST_URL: ${{ secrets.SONARQUBE_HOST }}
```

Note that this is identical for a commit and for a pull request. According to Sonar's documentation, the action code is able to tell the difference between the two actions.

**Comment by mc1arke** [comment]
@pcolmer the plugin does not currently provide auto-detection for Github Actions, so you'll need to manually specify the `sonar.branch` or `sonar.pullrequest` parameters. I see Sonarqube now also auto-detects Jenkins, CodeMagic, and Cirrus which the plugin does not have detection for either, so I'll raise a Pull Request to get these features added.

**Comment by pcolmer** [comment]
Thank you, @mc1arke! Really appreciate your support on this.

**Comment by pcolmer** [comment]
@mc1arke is it possible to test the commit you've made or do I need to wait for it to be incorporated into the released plugin?

**Comment by mc1arke** [comment]
Yes, see the artifacts section of https://github.com/mc1arke/sonarqube-community-branch-plugin/actions/runs/1635564337 for a snapshot build of that commit

**Comment by pcolmer** [comment]
@mc1arke Thank you.

I've incorporated the plugin and run a pull request against the `develop` branch of a project. I'm seeing two potential areas of concern within the SQ server:

1. The scan is recorded as a branch and not as a pull request.
2. The branch name is `evelop`.

I can see the following in the output from the scan action:

```Auto detected BRANCH configuration with source evelop using com.github.mc1arke.sonarqube.plugin.scanner.autoconfiguration.GithubActionsAutoConfigurer```

and, later on:

```INFO: Branch name: evelop```

Would you prefer me to log this information against #523 instead?

**Comment by mc1arke** [comment]
Could you add your findings to #526 please? That way it's easier for me to track what still needs performed on the PR

**Comment by mc1arke** [comment]
Released in 1.11.0
